src/main/java/at/ac/fhcampuswien/snake/board/GameBoard.java

```diff
@@ -0,0 +1,518 @@
+        return (snake.getSegments().get(0).getX() == food.getLocation().getX() &&
+                snake.getSegments().get(0).getY() == food.getLocation().getY());
+    }
+
+    /**
+     * This method centralizes the methods that draw walls in the game - the perimeter ones and walls that are inside
+     * the game area and used as obstacles, if any
+     *
+     * @param gc the current graphics context
+     */
+    private void drawWalls(GraphicsContext gc) {
+        drawPerimeterWalls(gc);
+        if (innerWall != null) drawInnerWalls(gc);
+    }
+
+    /**
+     * This method only draws the perimeter walls, without using the Wall class, to avoid overhead for collision check
+     * Collision with these walls should simply be a check if the snake is outside the gameboard area, subtracting
+     * the wall's thickness
+     *
+     * @param gc the current graphics context
+     */
+    private void drawPerimeterWalls(GraphicsContext gc) {
+        //Upper wall
+        for (int i = 0; i < GAME_BOARD_SIZE_MEDIUM; i += OBJECT_SIZE_MEDIUM) {
+            gc.drawImage(wallPattern, i, 0, OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM);
+        }
+
+        //Bottom wall - we subtract 2*OBJECT_SIZE_MEDIUM from the Y position to make it visible inside the Gameboard area
+        // and to account for the Menu bar
+        for (int i = 0; i < GAME_BOARD_SIZE_MEDIUM; i += OBJECT_SIZE_MEDIUM) {
+            gc.drawImage(wallPattern, i, GAME_BOARD_SIZE_MEDIUM - OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM);
+        }
+
+        //Left wall
+        for (int i = 0; i < GAME_BOARD_SIZE_MEDIUM; i += OBJECT_SIZE_MEDIUM) {
+            gc.drawImage(wallPattern, 0, i, OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM);
+        }
+
+        //Right wall - we subtract OBJECT_SIZE_MEDIUM from the X position to make it visible inside the Gameboard area
+        for (int i = 0; i < GAME_BOARD_SIZE_MEDIUM; i += OBJECT_SIZE_MEDIUM) {
+            gc.drawImage(wallPattern, GAME_BOARD_SIZE_MEDIUM - OBJECT_SIZE_MEDIUM, i, OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM);
+        }
+    }
+
+    private void drawInnerWalls(GraphicsContext gc) {
+        for (int i = 0; i < innerWall.getSegments().size(); i++) {
+            Position wallSegment = innerWall.getSegments().get(i);
+            gc.drawImage(wallPattern, wallSegment.getX(), wallSegment.getY(), OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM);
+        }
+    }
+
+    /**
+     * Initializes javafx events.
+     * <p>
+     * javafx events are events which are triggered by the user.
+     * For example a button click or a key press.
+     */
+    private void initializeEvents() {
+        // Checkout https://www.w3schools.com/java/java_lambda.asp for more information about lambdas.
+        // Basically the event gets passed as a parameter and can be used inside the parentheses.
+        gameBoardCanvas.setOnKeyPressed(event -> {
+            switch (event.getCode()) {
```

Comment:
Complete cases by adding the missing enum constants, or add a default case to this switch. Additionally, it is also long and complex.